internal/client/errors.go

```diff
@@ -26,9 +26,43 @@ func newAPIError(status int, bodyBuffer []byte, _ context.Context) ApiResponseEr
 	if len(bodyBuffer) == 0 {
 		return result
 	}
-	err := json.Unmarshal(bodyBuffer, &result)
-	if err != nil {
-		result.Message = fmt.Sprintf("failed to decode JSON body. reason: %s", err.Error())
+	// Errors *should* follow the expected format, but to be safe, we'll unmarshal into an
+	// interface and assert the fields we expect.
+	var e map[string]any
+
+	if err := json.Unmarshal(bodyBuffer, &e); err != nil {
+		result.Code = "EUNKNOWN"
+		result.Message = "There was an error, but the response from the server was not understood."
+		result.Errors = append(result.Errors, validationError{
+			Path:    "raw error body",
+			Message: string(bodyBuffer),
+		})
+		return result
+	}
+
+	if code, ok := e["code"].(string); ok {
+		result.Code = code
+	}
+	if message, ok := e["message"].(string); ok {
+		result.Message = message
+	}
+	if errors, ok := e["errors"].([]any); ok {
+		for _, err := range errors {
+			if e, ok := err.(map[string]any); ok {
+				vError := validationError{}
+				if path, ok := e["instancePath"].(string); ok {
+					vError.Path = path
+				}
+				if message, ok := e["message"].(string); ok {
+					vError.Message = message
+				}
+				result.Errors = append(result.Errors, vError)
+			}
+		}
+	} else if errorString, ok := e["errors"].(string); ok {
+		result.Errors = append(result.Errors, validationError{
+			Message: errorString,
+		})
```

Comment:
The use case for this whole thing is when `error` is a string, and not an array.  It seemed cleaner to do this than to try to try/catch unmarshal errors from the start, especially since we don't know if the unmarshal fails because of a bad body format or a bad `errors` format.

Reply:
We want the body to be consumed, so marshaling into an interface is the suggested approach that I saw.

Reply:
Makes sense, thanks!